### PR TITLE
ci: cache parsers for workflows/tests

### PR DESCRIPTION
**Pull request by MunifTanjim**

### CODE REVIEW

**Comment by clason** [comment]
we also need this for the `check-query-files-and-compilation` job?

**Comment by MunifTanjim** [comment]
> we also need this for the check-query-files-and-compilation job?

Yep, need to add it there too. Just checking if this works first. Waiting for it to pass so that I can retrigger.

**Review by theHamsta — APPROVED** [review]
And ccache in addition to that on Mac and Linux?

- https://cristianadam.eu/20200113/speeding-up-c-plus-plus-github-actions-using-ccache/
- https://github.com/hendrikmuhs/ccache-action

Another use for actions/cache would be `~/.cargo`, `~/.rustup` in tests.yml

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -33,8 +33,14 @@ jobs:
         run: |
           bash ./scripts/ci-install-${{ matrix.os }}.sh
 
+      - name: Setup Parsers Cache
+        id: parsers-cache
+        uses: actions/cache@v2
+        with:
+          path: ./parser/
+          key: ${{ matrix.os }}-parsers-v1-${{ hashFiles('./lockfile.json', './lua/nvim-treesitter/parsers.lua', './lua/nvim-treesitter/install.lua') }}
```
Comment (theHamsta) [inline .github/workflows/tests.yml:41]: could you add `shell_command_selectors.lua`?

**Review by theHamsta — APPROVED** [review]

**Comment by MunifTanjim** [comment]
Oh, this is already merged! Should I open another PR for the rest of the things? Or somebody is already taking care of it? @theHamsta

**Comment by theHamsta** [comment]
@MunifTanjim If you want to do it you can create another PR. The ccache stuff is probably not worth the effort but the Rust stuff certainly. I tried to compile a static binary on my repo but libmusl doesn't support dynamic loading (but could upload a build from Ubuntu 20.04 and download it in CI)

**Comment by clason** [comment]
@theHamsta you didn't wait for the PR to be actually finished ;) There was quite a bit missing.

So yes, please open a second PR for the rest (triggers, other workflow) -- although it'd help to have my PR merged first.

**Comment by MunifTanjim** [comment]
> it'd help to have my PR merged first

In that case, let's wait for it to get merged. I'll open another one later.

**Pull request merged**